config.go

```diff
@@ -5,28 +5,44 @@ import (
 )
 
 const (
-	apiURLv1                       = "https://api.openai.com/v1"
+	openaiAPIURLv1                 = "https://api.openai.com/v1"
 	defaultEmptyMessagesLimit uint = 300
+
+	azureAPIPrefix         = "openai"
+	azureDeploymentsPrefix = "deployments"
+)
+
+type APIType string
+
+const (
+	APITypeOpenAI  APIType = "OPEN_AI"
+	APITypeAzure   APIType = "AZURE"
+	APITypeAzureAD APIType = "AZURE_AD"
 )
 
 // ClientConfig is a configuration of a client.
 type ClientConfig struct {
 	authToken string
+	BaseURL   string
```

Comment:
Do we need to re-order fields here? The logic was to separate private `authToken` from others